File: Back-End/Web-POS.Api/Constants/DocumentTypeConstants.cs
namespace Api.Constants
{
    /// <summary>
    /// Single source of truth for DocumentType IDs and their number-series codes.
    /// IDs must match the DocumentType table; codes drive the YY-CCC-NNNNNN numbering scheme.
    /// </summary>
    public static class DocumentTypeConstants
    {
        // ── IDs (match DocumentType.Id in the database) ──────────────────────
        public const int Purchase       = 1;
        public const int Sales          = 2;
        public const int InventoryCount = 3;
        public const int Refund         = 4;
        public const int StockReturn    = 5;
        public const int LossAndDamage  = 6;
        public const int Proforma       = 7;

        // ── Series codes (used in document numbering: YY-{Code}-NNNNNN) ──────
        public const string PurchaseCode        = "100";
        public const string SalesCode           = "200";
        public const string InventoryCountCode  = "300";
        public const string RefundCode          = "220";
        public const string StockReturnCode     = "120";
        public const string LossAndDamageCode   = "400";
        public const string ProformaCode        = "230";
    }

    /// <summary>
    /// Single source of truth for DocumentCategory IDs.
    /// </summary>
    public static class DocumentCategoryConstants
    {
        public const int Expenses  = 1;
        public const int Sales     = 2;
        public const int Inventory = 3;
        public const int Loss      = 4;
    }

    /// <summary>
    /// Paid-status codes stored on the Document.PaidStatus column.
    /// </summary>
    public static class PaidStatusConstants
    {
        public const int Unpaid  = 0;
        public const int Paid    = 1;
        public const int Partial = 2;
        public const int Voided  = 99;
    }
}

File: Front-End/lib/document/document_type_constants.dart
/// Single source of truth for DocumentType IDs and series codes.
/// IDs must match the DocumentType table; codes drive YY-CCC-NNNNNN numbering.
abstract final class DocumentTypes {
  // ── IDs ───────────────────────────────────────────────────────────────────
  static const int purchase       = 1;
  static const int sales          = 2;
  static const int inventoryCount = 3;
  static const int refund         = 4;
  static const int stockReturn    = 5;
  static const int lossAndDamage  = 6;
  static const int proforma       = 7;

  // ── Series codes (used in document numbering: YY-{code}-NNNNNN) ──────────
  static const String purchaseCode        = '100';
  static const String salesCode           = '200';
  static const String inventoryCountCode  = '300';
  static const String refundCode          = '220';
  static const String stockReturnCode     = '120';
  static const String lossAndDamageCode   = '400';
  static const String proformaCode        = '230';
}

/// Single source of truth for DocumentCategory IDs.
abstract final class DocumentCategories {
  static const int expenses  = 1;
  static const int sales     = 2;
  static const int inventory = 3;
  static const int loss      = 4;
}

/// `DocumentType.StockDirection` — which way a type's lines move goods. Mirrors
/// the server's column; every stock rule reads it rather than a list of ids.
abstract final class StockDirections {
  static const int none       = 0;
  static const int intoStock  = 1;
  static const int outOfStock = 2;
}

/// Paid-status values stored on the document record.
abstract final class PaidStatus {
  static const int unpaid  = 0;
  static const int paid    = 1;
  static const int partial = 2;
  static const int voided  = 99;
}
